src/Windowing/Silk.NET.Windowing.Common/Internals/ViewImplementationBase.cs

```diff
@@ -178,15 +186,38 @@ public void DoRender()
 
         public void DoUpdate()
         {
-            // Check elapsed time
+            // Note this method is thread-safe and can be called from multiple threads.
+            var update = Update;
+            if (update is null)
+            {
+                // If there are no update callbacks then no need to track the update time.
+                return;
+            }
+
+            var updateTimestamp = Interlocked.Read(ref _updateTimestamp);
             var newTimestamp = Stopwatch.GetTimestamp();
-            if ((newTimestamp - _updateTimestamp) >= _updatePeriod)
+            var updatePeriod = Interlocked.Read(ref _updatePeriod);
+
+            /*
+             * The first check ensures that enough time has passed to perform an update, if it fails we return.
+             *
+             * If it succeeds then we try to update the timestamp, however, note that we may not 'win' if
+             * another thread updates the timestamp before us.  If we fail, we don't raise the Update event 
+             * as it will have been called by the winning thread.
+             *
+             * However, if updatePeriod <=0, there's an assumption that Update events will be called every time DoUpdate()
+             * is called.  As such we, only return if updatePeriod is positive and we failed.
+             */
+            if ((newTimestamp - updateTimestamp) < updatePeriod ||
```

Comment:
Normally we'd loop on a failure to update the timestamp. However, we can leverage the nature of the Update event to avoid looping by recognizing that (when `updatePeriod` > 0) there's no guarantee that Update is to be called for every `DoUpdate()` call, indeed we're more concerned with not calling _more than_ `UpdatesPerSecond`.  This allows a thread-safe implementation without lock usage.

Reply:
Seems fine.